packages/envd/internal/services/legacy/stream.go

```diff
@@ -0,0 +1,42 @@
+package legacy
+
+import (
+	"net/http"
+
+	"connectrpc.com/connect"
+	"github.com/rs/zerolog"
+)
+
+type streamConverter struct {
+	conn   connect.StreamingHandlerConn
+	logger *zerolog.Logger
+}
+
+func (s streamConverter) Spec() connect.Spec {
+	return s.Spec()
+}
+
+func (s streamConverter) Peer() connect.Peer {
+	return s.Peer()
+}
+
+func (s streamConverter) Receive(a any) error {
+	return s.conn.Receive(a)
+}
+
+func (s streamConverter) RequestHeader() http.Header {
+	return s.conn.RequestHeader()
+}
+
+func (s streamConverter) Send(a any) error {
+	a = maybeConvertValue(a)
```

Comment:
shouldn't this be done based on the header as in unary?

Reply:
I think it's different; at least the signature is different. We definitely need to test this though! I pushed up this implementation as a checkpoint, but I still have a lot of tests to write, and some manual work to verify.

Reply:
I maybe missed it, but can you point me to where the condition for "if to convert or not to convert" is in streaming done?

Reply:
Just implemented it. it's in interceptor.go, in the `WrapStreamingHandler` function.